src/wallet/wallet.cpp

```diff
@@ -2629,13 +2629,14 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
                 //  rediscover unknown transactions that were written with keys of ours to recover
                 //  post-backup change.
 
-                // Reserve a new key pair from key pool
+                // Reserve a new key pair from key pool. If it fails, provide a dummy
+                // destination in case we don't need change.
                 CTxDestination dest;
                 if (!reservedest.GetReservedDestination(dest, true)) {
-                    strFailReason = _("Can't generate a change-address key. Please call keypoolrefill first.").translated;
-                    return false;
+                    strFailReason = _("Transaction needs a change address, but we can't generate it. Please call keypoolrefill first.").translated;
                 }
                 scriptChange = GetScriptForDestination(dest);
+                assert(!dest.empty() || scriptChange.empty());
```

Comment:
It's not clear to me what this assert is checking for.

If `dest` is not empty, we got a change destination and so `scriptChange` is not empty. If dest is empty, then `scriptChange` will be empty. So this just covers all cases and can't fail?

Reply:
I can't remember why I wrote this, but I think the idea was to ensure `GetScriptForDestination()` returns an empty script for an empty destination. Perhaps that's overkill; I can drop it if there's any other changes needed in this PR.

Reply:
I guess it's fine, but it just seems like useless code that might be confusing to future readers. I would suggest that it be removed in a followup that touches this code.